imblearn/ensemble/easy_ensemble_generalization.py

```diff
@@ -0,0 +1,208 @@
+    >>>X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
+    >>>y = np.array([1, 1, 1, 2, 2, 2])
+    >>>eeg = EEG(voting='hard', random_state=0)
+    >>>eeg.fit(X,y)
+    >>>eeg.predict(X)
+    [1 1 1 2 2 2]
+    >>>
+    """
+
+    def __init__(self,
+                 base_estimator=None,
+                 base_sampler=None,
+                 n_estimators=5,
+                 voting='soft',
+                 random_state=None,
+                 n_jobs=1):
+
+        self.base_estimator = base_estimator
+        self.base_sampler = base_sampler
+        self.n_estimators = n_estimators
+        self.voting = voting
+        self.random_state = random_state
+        self.n_jobs = n_jobs
+
+    def _validate_estimator(self):
+        """Check the estimator and set the base_estimator_ attribute."""
+        super(EasyEnsembleGeneralization, self)._validate_estimator(
+            default=DecisionTreeClassifier())
+
+    def _validate_sampler(self):
+        """Check the sampler and set the base_sampler_ attribute."""
+
+        if self.base_sampler is not None:
+            self.base_sampler_ = self.base_sampler
+        else:
+            self.base_sampler_ = ROS()
+
+        if self.base_sampler_ is None:
+            raise ValueError("base_sampler cannot be None")
+
+    def fit(self, X, y, sample_weight=None):
+        """Build an ensemble of estimators from the training set (X, y).
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape = [n_samples, n_features]
+            The training input samples. Sparse matrices are accepted only if
+            they are supported by the base estimator.
+
+        y : array-like, shape = [n_samples]
+            The target values (class labels in classification, real numbers in
+            regression).
+
+        sample_weight : array-like, shape = [n_samples] or None
+            Sample weights. If None, then samples are equally weighted.
+            Note that this is supported only if the base estimator supports
+            sample weighting.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+
+        random_state = check_random_state(self.random_state)
+
+        self._validate_estimator()
+        self._validate_sampler()
+
+        random_state = check_random_state(self.random_state)
+        estimator_seeds = random_state.randint(MAX_INT, size=self.n_estimators)
+        sampler_seeds = random_state.randint(MAX_INT, size=self.n_estimators)
+
+        if not hasattr(self.base_sampler, 'random_state'):
+            ValueError('Base sampler must have a random_state parameter')
+
+        pipelines = []
+        seeds = zip(estimator_seeds, sampler_seeds)
+
+        for i, (estimator_seed, sampler_seed) in enumerate(seeds):
+
+            sampler = clone(self.base_sampler_)
+            sampler.set_params(random_state=sampler_seed)
+
+            if hasattr(self.base_estimator_, 'random_state'):
```

Comment:
We should reuse `make_estimator`? 
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/ensemble/base.py#L119